alibi_detect/utils/pytorch/kernels.py

```diff
@@ -52,43 +374,217 @@ def __init__(
             meaning that it should take in the tensors `x`, `y` and `dist` and return `sigma`.
         trainable
             Whether or not to track gradients w.r.t. `sigma` to allow it to be trained.
+        active_dims
+            Indices of the dimensions of the feature to be used for the kernel. If None, all dimensions are used.
+        feature_axis
+            Axis of the feature dimension.
         """
-        super().__init__()
-        if sigma is None:
-            self.log_sigma = nn.Parameter(torch.empty(1), requires_grad=trainable)
-            self.init_required = True
-        else:
-            sigma = sigma.reshape(-1)  # [Ns,]
-            self.log_sigma = nn.Parameter(sigma.log(), requires_grad=trainable)
-            self.init_required = False
-        self.init_sigma_fn = init_sigma_fn
+        super().__init__(active_dims, feature_axis)
+        self.parameter_dict['log-sigma'] = KernelParameter(
```

Comment:
General question: is it good practice to have hyphenated key values as parameters? @jklaise

Reply:
I'd stick to underscores as it's a standard.